src/monty/json.py

```diff
@@ -113,7 +111,9 @@ class B(A):
         mro = type(obj).mro()
     except TypeError:
         return False
-    return any(o.__module__ + "." + o.__name__ == ts for o in mro for ts in type_str)
+    return any(
+        o.__module__ + "." + o.__qualname__ == ts for o in mro for ts in type_str
```

Comment:
I believe using full qualified name to include the local scope might be better? @mturiansky In case I'm wrong here

Reply:
I don't claim to be an expert on any of this. I just came up with a working solution for the problem I was trying to address. If you think this is a better approach and leads to the same result, then no push-back from me. I'd need to refresh myself on what half of these variables/functions do to be able to make any meaningful comment.

Reply:
> I just came up with a working solution for the problem I was trying to address.

Thanks a ton for making everyone's life easier in the first place!

> If you think this is a better approach and leads to the same result

It just covers more edge cases but with unit test, it seems to do what we want it to do. Free feel to comment if I miss any type that should be tested :)

> I'd need to refresh myself on what half of these variables/functions do to be able to make any meaningful comment.

Take your time I would leave this as draft for a while

Reply:
I would mark it as ready for review now and feel free to comment :)